rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -2662,9 +2751,11 @@ RMW_Connext_Client::take_response(
   if (taken_msg) {
     request_header->request_id.sequence_number = rr_msg.sn;
 
+    /* (asorbini) although messages are correlated using the reply DataReader's GUID,
+       we report the writer's GUID to upper layers. */
     memcpy(
       request_header->request_id.writer_guid,
-      rr_msg.gid.data,
+      this->request_pub->gid()->data,
```

Comment:
I'm second guessing this change, and I'm thinking the value should be copied from `rr_msg.writer_gid.data`.

Reply:
I [reviewed `rmw_take_response()`](https://github.com/ros2/rmw/blob/83445be486deae8c78d275e092eafb4bf380bd49/rmw/include/rmw/rmw.h#L2053), and the `writer_guid` is supposed to match the guid returned by `rmw_send_request()`, so the value returned here is the correct one.